components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -666,6 +673,56 @@ internal sealed class SubscriptionWrapper : ISubscription {
                 return subscription;
             }
 
+            private void OnSubscriptionEventChanged(Subscription subscription, EventNotificationList notification, IList<string> stringTable) {
+                try {
+                    if (OnSubscriptionEventChange == null) {
+                        return;
+                    }
+                    if (notification == null) {
+                        _logger.Warning(
+                            "EventChange for subscription: {Subscription} having empty notification",
+                            subscription.DisplayName);
+                        return;
+                    }
+
+                    if (_currentlyMonitored == null) {
+                        _logger.Information(
+                            "EventChange for subscription: {Subscription} having no monitored items yet",
+                            subscription.DisplayName);
+                        return;
+                    }
+
+                    // check if notification is a keep alive
+                    var isKeepAlive = notification?.Events?.Count == 1 &&
```

Comment:
We can clean these null propagations as well. Null is checked for above. We could just as well extend that check with checking for a null or empty list of events.